ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/process/search_tree/Tree.java

```diff
@@ -169,4 +191,48 @@ static RaoResult buildOutput(Leaf rootLeaf, Leaf optimalLeaf) {
         raoResult.setPostOptimVariantId(optimalLeaf.getRaoResult().getPostOptimVariantId());
         return raoResult;
     }
+
+    private static void logOptimalLeaf(Leaf leaf, Crac crac) {
+        LOGGER.info(format("Optimal leaf - %s", generateLeafResults(leaf, crac)));
+    }
+
+    private static void logLeafResults(Leaf leaf, Crac crac) {
+        LOGGER.info(generateLeafResults(leaf, crac));
+    }
+
+    private static String generateLeafResults(Leaf leaf, Crac crac) {
+        String rangeActionResults = crac.getRangeActions()
+            .stream()
+            .map(rangeAction -> format("%s: %d",
+                rangeAction.getName(),
+                ((PstRangeResult) rangeAction.getExtension(RangeActionResultExtension.class)
+                    .getVariant(leaf.getRaoResult().getPostOptimVariantId()))
+                    .getTap(crac.getPreventiveState().getId())))
+            .collect(Collectors.joining(", "));
```

Comment:
why not also logs the network actions results ?

Reply:
It is done just below these lines.
`return` format("%s: minimum margin = %f (%s)",
            leaf.isRoot() ? "root leaf" : leaf.getNetworkActions().stream().map(NetworkAction::getName).collect(Collectors.joining(", ")),
            -leaf.getCost(crac),
            rangeActionResults);`